### ISSUE TITLE
Instructor input

### ISSUE THREAD

**Issue by JeremiahV93** [JeremiahV93/bear-watcher#18]
I could use some help understanding the directory structure of separating files of helpers, data, and components. Then if my functions are in the appropriate files, aka in the correct directory. I understand a lot of this is mostly workplace dependent and SOP, but I fail to understand what belongs in components vs helpers. Data seems a little straight forward but I could use help understanding this process of orginization.

**Comment by michaelclark2** [comment]
yeah, so this is the hard part about writing in a modular way, is deciding what goes where.  but the biggest difference I draw that differenciate helpers and components, is that helpers are methods and functions you use throughout the entire application, like the printToDom function.  and then a component would be code that directly relates to an element on the page.  so like the bear component and the river component.  when we talk more about components and do more projects, it will make more sense.  but this part was always a mind bender for me, and luckily there are gonna be senior devs who have a better opinion about where you should write your code so all of that will be figured out for you

**Comment by luketlancaster** [comment]
So to piggyback off what @michaelclark2 said:

components are going to be related to UI: getting values from the Dom (i.e. forms, ids, etc), printing things to the Dom, capturing user triggered events, etc.

data files are anything related to data manipulation (usually shortened to CRUD: Create, Read, Update, Delete): so all those functions live there. This is where it can kinda get murky...I like to keep all "events" out of my data function because they shouldn't care about user events: just taking in and giving out data

helpers are just what mc said: common code that can be used anywhere, so it's not domain specific.

----

looking over your code and everything looks to be in a good spot 👍
